components/users-table.tsx

```diff
@@ -1,36 +1,39 @@
 'use client'
 
 import { Radio, Table } from '@mantine/core'
-import { useAppDataStore } from '~/lib/hooks'
+import { type UserRow, useAppDataStore } from '~/lib/hooks'
 import { theme } from '~/lib/theme'
 import { atOrThrow } from '~/lib/util'
 
 interface UserRowCompProps {
   index: number
 }
 
-const selectRow = (index: number) => {
-  useAppDataStore.setState({ selectedIndex: index })
+const selectUser = (user: UserRow) => {
+  useAppDataStore.setState({ selectedUser: user })
 }
 
 const UserRowComp: React.FC<UserRowCompProps> = ({ index }) => {
   const data = useAppDataStore((state) => state.data)
-  const isSelected = useAppDataStore((state) => state.selectedIndex === index)
+  const user = atOrThrow(data?.rows ?? [], index)
+  const isSelected = useAppDataStore(
+    (state) => state.selectedUser === atOrThrow(data?.rows ?? [], index)
+  )
   if (!data) {
     return null
   }
-  const row = atOrThrow(data.rows, index)
+
   return (
     <Table.Tr
       key={index}
       bg={isSelected ? theme.colors.blue[0] : undefined}
-      onClick={() => selectRow(index)}
+      onClick={() => selectUser(user)}
     >
       <Table.Td>
-        <Radio aria-label='Select user' checked={isSelected} onChange={() => selectRow(index)} />
+        <Radio aria-label='Select user' checked={isSelected} onChange={() => selectUser(user)} />
       </Table.Td>
       {data.columns.map((col, index) => (
-        <Table.Td key={index}>{row[col] ?? ''}</Table.Td>
+        <Table.Td key={index}>{user[col] ?? ''}</Table.Td>
```

Comment:
Why did you migrate to `selectUser` from `selectRow`?

Reply:
> Why did you migrate to `selectUser` from `selectRow`? I think keeping track of the `index` is better.

I did that to make `warnings` global state.